main.js

```diff
@@ -1,46 +1,219 @@
-var selectDay = document.querySelector('.week');
-var schedule = document.querySelector('.event-viewer');
-var addEntry = document.querySelector('.addEntry');
-var modal = document.querySelector('.modal');
-var entryWeek = document.getElementById('entryWeek');
-var entrySubmit = document.getElementById('entrySubmit');
-var entryTime = document.getElementById('entryTime');
-var entryDescription = document.getElementById('entryDescription');
-var tableRow = document.querySelectorAll('tr');
-var tableBody = document.querySelector('tbody');
-
-selectDay.addEventListener('click', dayOfWeek);
-addEntry.addEventListener('click', entryModal);
-window.addEventListener('click', closeModal);
-entrySubmit.addEventListener('click', addTask);
-
-function dayOfWeek(event) {
-  if (event.target.className !== 'day') {
-    return;
+var schedule = {
+  Sunday: [],
+  Monday: [],
+  Tuesday: [],
+  Wednesday: [],
+  Thursday: [],
+  Friday: [],
+  Saturday: []
+};
+
+var todoList = {
+  addEntry: function(day, newTask) {
+    if (schedule[day] === undefined) {
+      console.log('schedule[day] is undefined');
+    }
+    var num = schedule[day].length;
+    schedule[day][num] = newTask;
+  },
+  updateEntry: function(day, position, newTask) {
+    schedule[day][position] = newTask;
+  },
+  deleteEntry: function (day, position) {
+    schedule[day].splice(position, 1);
   }
-  var sentence = event.target.textContent;
-  schedule.textContent = 'Scheduled Events for ' + sentence;
-}
-
-function entryModal() {
-  modal.classList.remove('hidden');
-}
-function closeModal(event) {
-  if (event.target === modal) {
-    modal.classList.add('hidden');
+};
+
+var handlers = {
+  addEntry: function() {
+    var modal = document.getElementById('modal-entry');
+    var entryWeek = document.getElementById('entryWeek');
+    var entryTime = document.getElementById('entryTime');
+    var entryDescription = document.getElementById('entryDescription');
+    var entrySubmit = document.getElementById('entrySubmit');
+
+    modal.classList.remove('hidden');
+
+    entrySubmit.addEventListener('click', function (event) {
+      event.preventDefault();
+
+      var newTask = {
+        day: entryWeek.value,
+        time: entryTime.value,
+        description: entryDescription.value,
+      };
+
+      todoList.addEntry(entryWeek.value, newTask);
```

Comment:
If the input is empty, then you're going to get undefined, just make sure its not undefined